pkgs/top-level/all-packages.nix

```diff
@@ -2456,8 +2456,9 @@ with pkgs;
 
   inherit (callPackages ../development/tools/ammonite {})
     ammonite_2_12
-    ammonite_2_13;
-  ammonite = if scala == scala_2_12 then ammonite_2_12 else ammonite_2_13;
+    ammonite_2_13
+    ammonite_3_2;
+  ammonite = if scala == scala_2_13 then ammonite_2_13 else ammonite_3_2;
```

Comment:
It couldn't hurt to support all versions of Scala that are in Nixpkgs. The only reason I didn't is that I'm a Nix novice, and couldn't figure out how to do a `case` statement, or a `elseif` statement. I guess you could do a map lookup, where the map is some mapping 

```
{"scala_2_13": "ammonite_2_13", 
 "scala_2_12": "ammonite_2_12",
...
```

And so on. I just don't know what the syntax for that would be. Feel free to suggest a method, and/or change this PR, or submit a new one.

Reply:
I'm not sure if I'm doing this right, but I just added a new commit which makes `scala_3` the default for both `ammonite` and `scala`. It only made sense to do them both, if you do one of them. I'm not a Scala programmer, but I have a feeling this is likely to be controversial, as it was for Python 3. Still, everything I can find online, like [the official scala docs](https://docs.scala-lang.org/), seems to have Scala 3 as the default, so it makes sense, even though it's likely to break some things. 

Still, leaving this here to generate discussion and/or improvements on this PR.

Reply:
I share your hesitation, but without Scala also on the latest version, it would mean that scala is version 2 (already outdated), and ammonite is version 3. So insalling `scala` and `ammonite` would mean you'd have different versions on each. So I think if it's not the right time to change `scala` to 3, then the default for `ammonite` should probably be `ammonite_2_13`. Although it still makes sense to me to have `scala` and `ammonite` both be on 3, which is where it seems to be from the official documentation. For example, for me as a Scala learner, all the books I've picked up have Scala 3 examples. So it'd be weird for another Scala beginner to install `scala` and `ammonite`, but then be unable to run any of the examples from the books.